tests/acl/test_acl_dscp_change.py

```diff
@@ -0,0 +1,654 @@
+    dstip = list(setUp_vnet[encap_type]['dest_to_nh_map'][vnet].keys())[0]
+    mask = ecmp_utils.HOST_MASK[ecmp_utils.get_payload_version(encap_type)]
+    dstPrefix = dstip + '/' + str(mask)
+    create_acl_rule(duthost, TABLE, '1', '9999', str(40), dstPrefix)
+    verify_acl_rules(setUp_vnet, duthost, encap_type, 40)
+    del_acl_rule(duthost, TABLE, '1')
+    del_acl_tables(duthost)
+
+
+def test_acl_rules_with_different_dscp(setUp, duthost, encap_type):
+    setUp_vnet = setUp
+    setup_acl_tables(duthost, setUp_vnet)
+    if 'v4_in' in encap_type:
+        TABLE = TABLE_V4
+    else:
+        TABLE = TABLE_V6
+    vnet = list(setUp_vnet[encap_type]['vnet_vni_map'].keys())[0]
+    dstip = list(setUp_vnet[encap_type]['dest_to_nh_map'][vnet].keys())[0]
+    mask = ecmp_utils.HOST_MASK[ecmp_utils.get_payload_version(encap_type)]
+    dstPrefix = dstip + '/' + str(mask)
+
+    # Create multiple rules with different priorities, rule IDs, and SRC_IP/SRC_IPV6 fields
+    if TABLE == TABLE_V4:
+        rules = [
+            {'rule_id': '1', 'priority': '100', 'dscp_action': '10', 'match': '"SRC_IP":"170.170.170.1/32"'},
+            {'rule_id': '2', 'priority': '200', 'dscp_action': '20', 'match': '"SRC_IP":"170.170.170.2/32"'},
+            {'rule_id': '3', 'priority': '300', 'dscp_action': '30', 'match': '"SRC_IP":"170.170.170.3/32"'},
+            {'rule_id': '4', 'priority': '400', 'dscp_action': '40', 'match': '"SRC_IP":"170.170.170.4/32"'},
+            {'rule_id': '5', 'priority': '500', 'dscp_action': '50', 'match': '"SRC_IP":"170.170.170.5/32"'}
+        ]
+    else:
+        rules = [
+            {'rule_id': '1', 'priority': '100', 'dscp_action': '10',
+             'match': '"SRC_IPV6":"9999:AAAA:BBBB:CCCC:DDDD:EEEE:FFFF:1/128"'},
+            {'rule_id': '2', 'priority': '200', 'dscp_action': '20',
+             'match': '"SRC_IPV6":"9999:AAAA:BBBB:CCCC:DDDD:EEEE:FFFF:2/128"'},
+            {'rule_id': '3', 'priority': '300', 'dscp_action': '30',
+             'match': '"SRC_IPV6":"9999:AAAA:BBBB:CCCC:DDDD:EEEE:FFFF:3/128"'},
+            {'rule_id': '4', 'priority': '400', 'dscp_action': '40',
+             'match': '"SRC_IPV6":"9999:AAAA:BBBB:CCCC:DDDD:EEEE:FFFF:4/128"'},
+            {'rule_id': '5', 'priority': '500', 'dscp_action': '50',
+             'match': '"SRC_IPV6":"9999:AAAA:BBBB:CCCC:DDDD:EEEE:FFFF:5/128"'}
+        ]
+
+    for rule in rules:
+        create_acl_rule(duthost,
+                        TABLE,
+                        rule['rule_id'],
+                        rule['priority'],
+                        rule['dscp_action'],
+                        dstPrefix, rule['match'])
+    for rule in rules:
+        verify_acl_rules(setUp_vnet,
+                         duthost,
+                         encap_type,
+                         int(rule['dscp_action']),
+                         rule['match'])
+
+    # Delete all rules
+    for rule in rules:
```

Comment:
Suggest handling the cleanup in a fixture to ensure it's always called even if exception is raised.